### PR TITLE
Round cell size to nearest integer [NSPECT-61]

### PR DESCRIPTION
**Pull request by idtodd**
Output raster size is rounded to the nearest integer. Added an option to set the output raster resolution rather than always adopting it from the reference raster.

### CODE REVIEW

File: components/align_rasters/align_rasters.py [hunk 1]
```diff
@@ -68,6 +71,16 @@ def initAlgorithm(self, config=None):
                 defaultValue=100,
             )
         )
+        self.addParameter(
+            QgsProcessingParameterNumber(
+                self.rasterCellSize,
+                "Output Cell Size",
+                optional=True,
+                type=QgsProcessingParameterNumber.Integer,
```
Comment (ar-siddiqui) [inline components/align_rasters/align_rasters.py]: We should allow non-integers here as well. If users wants to put integer value they can always do that.

File: components/align_rasters/align_rasters.py [hunk 2]
```diff
@@ -68,6 +71,16 @@ def initAlgorithm(self, config=None):
...
+                minValue=1,
```
Comment (ar-siddiqui) [inline components/align_rasters/align_rasters.py]: min value should be 0 or just greater than 0 (0.0000000000000000000000001) something like this not sure if it is a good idea or not.

**Review by ar-siddiqui — CHANGES_REQUESTED** [review]
@idtodd please address these comments

File: components/align_rasters/align_rasters.py (hunk 2 above)
Comment (idtodd) [reply components/align_rasters/align_rasters.py]: Set to 0.000001 Since that's the smallest where it will show as non-zero in the GUI.

**Review by ar-siddiqui — APPROVED** [review]

**Pull request merged**